pkg/odo/cli/logs/logs.go

```diff
@@ -133,17 +135,40 @@ func (o *LogsOptions) Run(ctx context.Context) error {
 	}
 
 	uniqueContainerNames := map[string]struct{}{}
+	errChan := make(chan error, len(containersLogs))
+	wg := sync.WaitGroup{}
+	var mu sync.Mutex
 	for _, entry := range containersLogs {
 		for container, logs := range entry {
 			uniqueName := getUniqueContainerName(container, uniqueContainerNames)
 			uniqueContainerNames[uniqueName] = struct{}{}
-			err = printLogs(uniqueName, logs, o.out)
-			if err != nil {
-				return err
-			}
+			wg.Add(1)
+			go func(follow bool) {
+				defer wg.Done()
+				colour := log.ColorPicker()
+				if !follow {
+					// ensure that only one go routine does printLogs at a time; this is helpful when logs are longer
+					// than just a few lines in that logs for each container are printed before starting printing those
+					// of the next container
+					mu.Lock()
+				}
+				err = printLogs(uniqueName, logs, o.out, colour)
+				if err != nil {
+					errChan <- err
+				}
+				if !follow {
+					mu.Unlock()
+				}
+			}(o.follow)
 		}
 	}
-
+	select {
+	case err := <-errChan:
+		return err
+	default:
+		// do nothing
```

Comment:
Without this `default` block, `odo logs` (without follow) will be stuck/blocked on getting something from `errChan` channel and the command won't exit.

Reply:
> Without this `default` block, `odo logs` (without follow) will be stuck/blocked on getting something from `errChan` channel and the command won't exit.

Can you add this comment in the code?

Reply:
Sure. Added.